readthedocs/proxito/views/serve.py

```diff
@@ -599,11 +599,15 @@ def _get_custom_404_page(self, request, project, version=None):
                     version_slug_404=version_404.slug,
                     storage_filename_path=storage_filename_path,
                 )
-                resp = HttpResponse(
-                    build_media_storage.open(storage_filename_path).read()
-                )
-                resp.status_code = 404
-                return resp
+                try:
+                    content = build_media_storage.open(storage_filename_path).read()
+                except FileNotFoundError:
+                    log.warning(
+                        "File not found in storage. File out of sync with DB.",
+                        file=storage_filename_path,
+                    )
+                    return None
+                return HttpResponse(content, status=404)
```

Comment:
```suggestion
                try:
                    content = build_media_storage.open(storage_filename_path).read()
                    return HttpResponse(content, status=404)
                except FileNotFoundError:
                    log.warning(
                        "File not found in storage. File out of sync with DB.",
                        file=storage_filename_path,
                    )
                    return None
```

That way is more clear where `content=` comes from.